trie-db/src/lookup.rs

```diff
@@ -118,7 +144,7 @@ where
 				// check if new node data is inline or hash.
 				match next_node {
 					NodeHandle::Hash(data) => {
-						hash = decode_hash::<L::Hash>(data)
+						self.hash = decode_hash::<L::Hash>(data)
```

Comment:
why do we even assign it if we are going to break anyway ?
```suggestion
						decode_hash::<L::Hash>(data)
```